### PR TITLE
norm logging per param

### PR DESCRIPTION
**Pull request by haeggee**
with the flag `log-params-norm-per-param`, allows to log the l2 norms of all parameters. i have not done throughput tests yet (we should), but his will be helpful for debugging when the loss explodes to potentially pinpoint the source 

EDIT: i had tried with DP and TP, where the global norm matches (up to some precision) the summed up norms of the individual norms calculated by the new function. also, i removed the incorrect lines that i myself had added to the global norm computation -- not sure what was going wrong before when i was testing it, but it is working fine now

### CODE REVIEW

File: megatron/training/arguments.py [hunk 1]
```diff
@@ -1162,6 +1162,8 @@ def _add_logging_args(parser):
 
     group.add_argument('--log-params-norm', action='store_true',
                        help='If set, calculate and log parameters norm.')
+    group.add_argument('--log-params-norm-per-param', action='store_true',
+                       help='If set, calculate and log parameters norm.')
```
Comment (TJ-Solergibert) [inline megatron/training/arguments.py:1166]: Same help comment as `--log-params-norm`!

File: megatron/training/training.py [hunk 2]
```diff
@@ -1017,6 +1018,13 @@ def training_log(loss_dict, total_loss_dict, learning_rate, decoupled_learning_r
                               args.consumed_train_samples)
             if wandb_writer:
                 wandb_writer.log({'params-norm': params_norm}, iteration)
```
Comment (TJ-Solergibert) [inline megatron/training/training.py:1020]: Do you think we can add this entry to `params-norm/` too? Something like `params-norm/global`?

Comment (TJ-Solergibert) [reply megatron/training/training.py:1020]: After observing a [wandb](https://wandb.ai/haeggee/TheMeg-Clariden/runs/v8zc671g?nw=nwuserhaeggee) project, let's keep it like it is

**Review by TJ-Solergibert — APPROVED** [review]
👌
